lib/_http_outgoing.js

```diff
@@ -426,6 +430,37 @@ OutgoingMessage.prototype.getHeader = function getHeader(name) {
 };
 
 
+// Returns an array of the names of the current outgoing headers.
+OutgoingMessage.prototype.getHeaderNames = function getHeaderNames() {
+  return (this._headers ? Object.keys(this._headers) : []);
+};
+
+
+// Returns a shallow copy of the current outgoing headers.
+OutgoingMessage.prototype.getHeaders = function getHeaders() {
+  const headers = this._headers;
+  const ret = new OutgoingHeaders();
+  if (headers) {
+    const keys = Object.keys(headers);
+    for (var i = 0; i < keys.length; ++i) {
+      const key = keys[i];
+      const val = headers[key][1];
+      ret[key] = val;
+    }
+  }
+  return ret;
+};
+
+
+OutgoingMessage.prototype.hasHeader = function hasHeader(name) {
+  if (typeof name !== 'string') {
+    throw new TypeError('"name" argument must be a string');
+  }
+
+  return !!(this._headers && this._headers[name.toLowerCase()]);
```

Comment:
A `has` function returning true for something which is actually not there doesn't feel right. Strictly speaking, those inherited properties are not *headers*.